drivers/timer: stm32: Convert configuration from Kconfig to device tree

erwango opened the pull request:
Use device tree to configure lptim driver with following benefits:
- Allow usage of instances other than lptim1 if needed
- Simplify driver code, specially lptim source clock selection (thanks to use of dt domain clock)

Make some related clean up in board description files.

erwango committed:
drivers: timer: stm32: Use dt instance for LPTIM base address /IRQ

Start converting LPTIM driver to device tree based configuration and
support of other instances.
First: get base address and IRQ using dt instance

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers: timer: stm32: Use dt instance for LPTIM bus clock

Continue conversion of LPTIM driver to device tree based configuration.
Get clock configuration from device tree and use clock_control API for
bus clock configuration

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers: timer: stm32: Use dt to configure LPTIM domain clock

Instead of relying on Kconfig, use dt inputs to configure LPTIM domain
clck (LSI/lSE).
Clock control dedicated APIs are used for configuration and get the
frequency of domain clock in use.
Constants macros used previously to store frequency and time base are
converted to static global variables.

Some code was set up specifically to keep compatibility with targets
that still use Kconfig to configure domain clock. This will be removed
after a deprecation period.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: stm32: Define lptim domain clock using device tree

Convert lptim configuration from Kconfig to device tree.
Note that some boards were using Kconfig default value (LSI),
which now has to be made explicit using device tree.

If not done already, enable the required clock node (lse/lsi).

Clean up related Kconfig bits when required.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: stm32: Remove lptim node when not used in tree

These boards enable lptim node, but don't propose 'power-states'
description, making it useless and not testable regarding configuration
of existing, in tree, PM tests.

Clean up this definition.

They will be put back along with 'power-states', when possible to test.


Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
CI failed due to `tests/net/` failing on mps2 target:
```
tests/net/socket/tcp/net.socket.tcp (mps2_an385:tests/net/socket/tcp/net.socket.tcp) failed
```

gmarull commented on drivers/timer/stm32_lptim_timer.c:
```diff
@@ -27,6 +27,9 @@
 #define LPTIM (LPTIM_TypeDef *) DT_INST_REG_ADDR(0)
 
+static const struct stm32_pclken lptim_clk[] = STM32_DT_INST_CLOCKS(0);
+static const struct device *clk_ctrl = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
```
nit: check for readiness if https://github.com/zephyrproject-rtos/zephyr/pull/48787 goes first

erwango replied:
Thanks, done

erwango committed:
boards: stm32l562e_dk: Factorize lp related nodes

Low power related nodes don't depend on Trust Zone activation.
Factorize into -common.dtsi when possible.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ABOSTM commented on drivers/timer/stm32_lptim_timer.c:
```diff
@@ -281,44 +291,26 @@ static int sys_clock_driver_init(const struct device *dev)
-	while (!LL_RCC_LSE_IsReady()) {
-		/* Wait for LSE ready */
+	/* A 32bit value cannot exceed 0xFFFFFFFF/lptim_time_base counting cycles.
+	 * This is for example about of 65000 x 2000ms when clocked by LSI
```
The original comment should not be moved,
And to explain the 0xFA00 value:
`When clocked by LSI @32K:   32000Hz x 2s = 65000 = 0xFA00`

erwango committed:
doc: release-notes-3.2: Log STM32 LPTIM Kconfig as deprecated

Log deprecation of Kconfig based LPTIM domain clock configuration.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

FRASTM commented on boards/arm/nucleo_wb55rg/nucleo_wb55rg.dts:
```diff
@@ -179,6 +179,8 @@
 
 &lptim1 {
+	clocks = <&rcc STM32_CLOCK_BUS_APB1 0x80000000>,
+		 <&rcc STM32_SRC_LSE LPTIM1_SEL(1)>;
```
STM32_SRC_LSE LPTIM1_SEL(**3**)

ABOSTM reviewed (approved):
LGTM
